app/services/network_topology_service.rb

```diff
@@ -0,0 +1,124 @@
+class NetworkTopologyService < TopologyService
+  def initialize(provider_id)
+    @provider_id = provider_id
+    @providers = retrieve_providers(ManageIQ::Providers::NetworkManager, @provider_id)
+    # @entity = retrieve_entity(entity_id, entity_type)
+    @entity = @providers.first
+  end
+
+  def entity_type(entity)
+    if entity.kind_of?(CloudNetwork)
+      entity.class.base_class.name.demodulize
+    else
+      entity.class.name.demodulize
+    end
+  end
+
+  def build_topology
+    topo_items = {}
+    links = []
+
+    entity_relationships = {
+      :NetworkManager => {
```

Comment:
imo you don't have to base your PR on the recursive topology PR (#7363) for the release, it can work without it as well. #7363 will wait a while for #6115 because it is a preparation for entity based topology, so it assumes there's a single entity (whether a provider or other entity) to build the topology for, while the current way of work is a separate tab that can receive multiple ids to build topology for.